lib/index.js

```diff
@@ -21,22 +23,52 @@ exports.plugin = {
         const p2p = container.resolvePlugin('p2p')
 
         container.resolvePlugin('p2p').getMonitor().broadcastTransactions = async transactions => {
-            const originalLength = transactions.length
-
             // Filter transactions, based on the public keys in the config
-            transactions = transactions.filter(transaction => !options.publicKeys.includes(transaction.data.senderPublicKey))
+            transactions = transactions.reduce((transactions, transaction) => {
+                if (options.publicKeys.includes(transaction.data.senderPublicKey)) {
+                    transactions.filtered.push(transaction)
+                } else {
+                    transaction.unfiltered.push(transaction)
+                }
+                return transactions
+            }, {
+                filtered: [],
+                unfiltered: [],
+            })
 
-            const lengthDiff = originalLength - transactions.length
-            if (lengthDiff) {
-                logger.debug(`[VANIR] Filtered ${pluralize(
-                    'transaction',
-                    lengthDiff,
-                    true,
-                )} to self-forge`)
-            }
+            if (transactions.filtered.length) {
+                let delegates = options.delegates || []
+
+                if (!Array.isArray(delegates)) {
+                    delegates = [delegates]
+                }
 
-            if (!transactions.length) {
-                return;
+                let hasActiveDelegates = !delegates.length
+
+                if (delegates.length) {
+                    const database = container.resolvePlugin('database')
+                    const activeDelegates = (await database.getActiveDelegates()).map(delegate => {
+                        return delegate.username || delegate.attributes.delegate.username
+                    })
+
+                    hasActiveDelegates = delegates.some(delegate => activeDelegates.includes(delegate))
+                }
+
+                // Check if any of the configured delegates is currently forging
+                if (options.overrideNotForging || hasActiveDelegates) {
```

Comment:
when `options.overrideNotForging` is true, it's unnecessary to do the whole delegate lookup as the `hasActiveDelegates` value will be redundant. Might be better to move this to the start of the function and handle the delegate lookup only if this value is not set to `true`

Reply:
That option can be removed. It was needed for the previous approach where the delegates would be retrieved automatically.

Reply:
Removed in d4a1659.